### PR TITLE
Support for ot-rcp-spi

### PR DESCRIPTION
**Pull request by parag-silabs**
- Add a new project file (openthread-efr32-rcp-spi.slcp) to generate openthread-efr32-rcp-spi and openthread-efr32-rcp-spi-mbedtls libs.
- Add a patch file to apply over gecko_sdk for ot-rcp-spi target to generate libs and reset head back after generation.
- Update build script to build 'ot-rcp-spi' image.
- Re-organize build directories to compile UART and SPI ot-rcp targets into respective dirs.

### CODE REVIEW

File: src/platform_projects/openthread-efr32-rcp.slcp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 project_name: openthread-efr32-rcp
```
Comment (romacdon) [inline src/platform_projects/openthread-efr32-rcp.slcp:1]: Should we rename this file/project to `openthread-efr32-rcp-uart.slcp` to be consistent with the new `openthread-efr32-rcp-spi.slcp` file/project.

File: src/platform_projects/openthread-efr32-rcp-spi.slcp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+project_name: openthread-efr32-rcp-spi
+label: ot-efr32
+package: OpenThread
+description: This is the platform project that will enable the ot-efr32 repo on GitHub to generate a CMake library for EFR32 RCPs which communicate over SPI.
+category: OpenThread Examples
+quality: production
+
+component:
+  - id: brd4001a
```
Comment (romacdon) [inline src/platform_projects/openthread-efr32-rcp-spi.slcp]: We don't need/want the brd4001a. This was removed from all the slcp files in the GSDK.  Probably need to fix the uart version as well.

File: src/platform_projects/openthread-efr32-rcp.slcp (hunk 1 above)
Comment (romacdon) [inline src/platform_projects/openthread-efr32-rcp.slcp]: Should we rename this file/project to `openthread-efr32-rcp-uart.slcp` to be consistent with the new `openthread-efr32-rcp-spi.slcp` file/project.

**Review by romacdon — CHANGES_REQUESTED** [review]
Just a few small changes requested.  Nice job overall.

File: src/platform_projects/openthread-efr32-rcp.slcp (hunk 1 above)
Comment (lmnotran) [reply src/platform_projects/openthread-efr32-rcp.slcp]: Good suggestion. @parag-silabs you'll need to change this too https://github.com/openthread/ot-efr32/blob/main/CMakeLists.txt#L67

File: src/platform_projects/openthread-efr32-rcp-spi.slcp (hunk 2 above)
Comment (parag-silabs) [reply src/platform_projects/openthread-efr32-rcp-spi.slcp]: Yes, cleaned up from all the .slcps.

File: src/platform_projects/openthread-efr32-rcp.slcp (hunk 1 above)
Comment (parag-silabs) [reply src/platform_projects/openthread-efr32-rcp.slcp]: Yeah good point. Renamed the project file and made required changes to build the project.

**Pull request closed without merging**